python_backend/api.py

```
@@ -856,6 +865,942 @@ async def generate_slide_previews_endpoint(
+            "marked_failed": marked_failed,
+            "ready_for_retry": ready_for_retry,
+            "effective_limit": effective_limit,
+            "details": details
+        }
+
+    except Exception as e:
+        logger.exception("Recovery run failed")
+        raise HTTPException(500, str(e))
+
+
+async def _process_single_job(job: dict, semaphore: asyncio.Semaphore) -> dict:
```

Ruff (0.14.13)

Undefined name `asyncio`

(F821)